pandas/tests/test_base.py

```diff
@@ -972,6 +972,34 @@ def test_memory_usage(self):
             diff = res_deep - sys.getsizeof(o)
             self.assertTrue(abs(diff) < 100)
 
+    def test_searchsorted(self):
+        # See gh-12238
+        for o in self.objs:
+            # sanity check
+            index = np.searchsorted(o, max(o))
+            self.assertTrue(0 <= index <= len(o))
+
+            msg = ("Non-default values for \'sorter\' are only "
+                   "accepted for Categorical and Series objects "
+                   "storing ndarray, Categorial, or Series data.")
+            with tm.assertRaisesRegexp(ValueError, msg):
+                np.searchsorted(o, max(o), sorter=range(len(o)))
+
+        # check the Series and Categorical classes separately from
+        # those contained in the 'self.objs' array due to their
+        # different behaviour when a non-None sorter parameter
+        # value is passed into their searchsorted method
+        s = pd.Series(np.arange(10))
+        c = pd.Categorical(np.arange(10))
+
+        for o in (c, s):
+            # sanity check
+            index = np.searchsorted(o, max(o))
+            self.assertTrue(0 <= index <= len(o))
+
+            with tm.assert_produces_warning(DeprecationWarning):
```

Comment:
remove this (and FYI we don't ever use ``DeprecationWarning``). We use ``FutureWarning``